### ISSUE TITLE
Big DNS updates

### ISSUE THREAD

**Issue by virtser** [github/octodns#527]
Anyone tried using OctoDNS to update thousands of records at a time?
We need it for our production environment gradual migration and we wonder if OctoDNS will do the trick.
We already found out that there is a MAX_SAFE_UPDATE_PCENT fixed threshold of 30%, but this can be fixed by passing it from config if required.
We are using NS1 and Constellix as providers.

**Comment by yzguy** [comment]
It should be fine, what you'll likely run into is rate limiting from the provider.

What you might want to do is dump the zone you're trying to migrate, (eg. example.com), change the domain to something new (exampletestzone.com), rename files as necessary, etc and put it in your config, then run it against each provider.

This would allow you to set up a parallel/unused zone with the same amount of records. When you're done you can just delete the test zone.

Potentially as well, because I know both those providers can be secondary DNS servers is just set them up Primary/Secondary and then you only have to push to one. I'd recommend NS1 because their have a much easier to use API.

**Comment by ross** [comment]
Nothing inherently prevents it. I don't imagine 1000's would specifically be an issue for octoDNS itself and I've tested that sort of numbers in the past looking into performance issues with specific providers etc. You may run into rate-limiting issues (especially with NS1 where there's on-going work/tweaks around that.) I haven't personally used Constellix so can't say there.

**Comment by brianeclow** [comment]
I have used octoDNS for (now) multiple provider migrations and for sweeping TTL updates because one origin provider had a silly default. The only hiccup I hit was the 30% change, which the `--force` flag overrides. However, I only used after vetting the changes, as it is a hammer with dangerous consequences. And the second is the API rate limiting. Luckily, with the mass TTL updates, the provider was feeling generous, and didn't nerf me. However, some of the mass importing/migrations did ruffle some feathers because of the extra API operations.
